### ISSUE TITLE
Rule idea: Remove unused interpolation

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#8419]
Suggest to remove $ or add missing interpolation:

Useless $ adds confusion. Or causes a bug when {} are missing.
```
var count = 42;
var value1 = $"Some {count}"; // Compliant
var value2 = $"Some count";  // Noncompliant
var value3 = $$"Some count"; // Noncompliant
var value4 = $$$"Some count"; // Noncompliant
var value5 = $"""Some count"""; // Noncompliant
var value6 = $"""
    Some count
    """; // Noncompliant
var value7 = $$"""
    Some count
    """; // Noncompliant
```
and so on

**Comment by jilles-sg** [comment]
An interpolation without `{}` may still be useful if it corresponds to a `FormattableString` or interpolation handler (as opposed to a string).

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jilles-sg,

Thank you for the comment, it will surely help us reduce false positives. I can clearly see the point with the interpolation handler. 

Can you please elaborate a little on the `FormattbleString` scenario?

**Comment by jilles-sg** [comment]
`FormattableString` is a fancy tuple of a format string and an array of arguments. It allows doing basically the same as interpolation handlers, but works on older versions of .NET and .NET Framework and does more allocations. If an interpolated string is used where a `FormattableString` or `IFormattable` is expected, a `FormattableString` is created.

Basically, if an interpolation without `{}` corresponds to something other than a string, removing the `$` might change the behaviour of the code or cause it to stop compiling.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1639](https://sonarsource.atlassian.net/browse/NET-1639)

[NET-1639]: https://sonarsource.atlassian.net/browse/NET-1639?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
